ops/services/alerts/app_service_metrics/main.tf

```diff
@@ -93,41 +93,30 @@ resource "azurerm_monitor_smart_detector_alert_rule" "failure_anomalies" {
   }
 }
 
-resource "azurerm_monitor_metric_alert" "http_2xx_failed_requests" {
+resource "azurerm_monitor_scheduled_query_rules_alert" "http_2xx_failed_requests" {
   name                = "${var.env}-api-2xx-failed-requests"
   description         = "${local.env_title} HTTP Server 2xx Errors (where successful request == false) >= 10"
+  location            = data.azurerm_resource_group.app.location
   resource_group_name = var.rg_name
-  scopes              = [var.app_insights_id]
-  frequency           = "PT1M"
-  window_size         = "PT5M"
   severity            = var.severity
+  frequency           = 5
+  time_window         = 5
   enabled             = contains(var.disabled_alerts, "http_2xx_failed_requests") ? false : true
 
-  criteria {
-    aggregation      = "Count"
-    metric_name      = "requests/count"
-    metric_namespace = "Microsoft.Insights/Components"
-    operator         = "GreaterThanOrEqual"
-    threshold        = 10
+  data_source_id = var.app_insights_id
 
-    dimension {
-      name     = "request/success"
-      operator = "Include"
-      values   = ["False"]
-    }
+  query = <<-QUERY
+requests
+| where toint(resultCode) between (200 .. 299) and success == false and timestamp > ago(5m)
```

Comment:
**question:** This makes sense! Might want this to be `>= ago(5m)`? Or does it seem nigh-impossible for us to miss a failure on exact timestamp equality

Reply:
Hmm. Worst case scenario is that if that happened, we'd catch it in the next window (5 minutes later) but it can't hurt to safeguard! Good idea.